src/core/ApplicationDescription.h

```diff
@@ -150,8 +150,7 @@ class ApplicationDescription {
     bool useUnlimitedMediaPolicy() const { return m_useUnlimitedMediaPolicy; }
     const std::string& locationHint() const { return m_locationHint; }
 
-    int surfaceRole() const { return m_surface_role; }
-    int panelType() const { return m_panel_type; }
+    const struct agl_shell_surface *getShellSurface() { return m_surface; } 
```

Comment:
I think the right refactor is providing an abstract shell surface object (or shell surface properties). See below.